src/pages/instances/forms/DuplicateInstanceForm.tsx

```diff
@@ -0,0 +1,278 @@
+import { FC, useState } from "react";
+import { LxdInstance } from "types/instance";
+import { useFormik } from "formik";
+import { useToastNotification } from "context/toastNotificationProvider";
+import {
+  ActionButton,
+  Button,
+  Form,
+  Input,
+  Modal,
+  Select,
+} from "@canonical/react-components";
+import * as Yup from "yup";
+import { createInstance, fetchInstances } from "api/instances";
+import { isClusteredServer } from "util/settings";
+import { useSettings } from "context/useSettings";
+import { useQuery } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { fetchStoragePools } from "api/storage-pools";
+import { fetchClusterMembers } from "api/cluster";
+import { Link } from "react-router-dom";
+import { instanceNameValidation } from "util/instances";
+import { fetchProjects } from "api/projects";
+import { LxdDiskDevice } from "types/device";
+import InstanceLink from "pages/instances/InstanceLink";
+import { useEventQueue } from "context/eventQueue";
+
+interface Props {
+  instance: LxdInstance;
+  close: () => void;
+}
+
+export interface LxdInstanceDuplicate {
+  instanceName: string;
+  targetProject: string;
+  targetClusterMember?: string;
+  targetStoragePool: string;
+  allowInconsistent: boolean;
+  instanceOnly: boolean;
+  stateful: boolean;
+}
+
+const DuplicateInstanceForm: FC<Props> = ({ instance, close }) => {
+  const toastNotify = useToastNotification();
+  const { data: settings } = useSettings();
+  const isClustered = isClusteredServer(settings);
+  const controllerState = useState<AbortController | null>(null);
+  const eventQueue = useEventQueue();
+
+  const { data: projects = [], isLoading: projectsLoading } = useQuery({
+    queryKey: [queryKeys.projects],
+    queryFn: fetchProjects,
+  });
+
+  const { data: storagePools = [], isLoading: storagePoolsLoading } = useQuery({
+    queryKey: [queryKeys.storage],
+    queryFn: () => fetchStoragePools(instance.project),
+  });
+
+  const { data: clusterMembers = [], isLoading: clusterMembersLoading } =
+    useQuery({
+      queryKey: [queryKeys.cluster],
+      queryFn: fetchClusterMembers,
+      enabled: isClustered,
+    });
+
+  const notifySuccess = (instanceName: string, instanceProject: string) => {
+    const newInstance = (
+      <Link to={`/ui/project/${instanceProject}/instance/${instanceName}`}>
+        {instanceName}
+      </Link>
+    );
+    toastNotify.success(
+      <>
+        Created instance <b>{newInstance}</b>.
+      </>,
+    );
+  };
+
+  const getDuplicatedInstanceName = (instance: LxdInstance): string => {
+    const { data: instances = [] } = useQuery({
+      queryKey: [queryKeys.instances],
+      queryFn: () => fetchInstances(instance.project),
+    });
+
+    const instanceNames = instances.map((instance) => instance.name);
+    const newInstanceName = instance.name + "-duplicate";
```

Comment:
there is one condition we need to check here, is to ensure `instance.name + "-duplicate"` does not exceed 63 characters thus violating lxd instance naming constraint. You can use `truncateInstanceName` from `util/instances` for this like `newInstanceName = truncateInstanceName(instance.name, "-duplicate")`. Then rest of the logic can remain the same.